CME on new server

Oopsy.

```
java.util.ConcurrentModificationException
java.util.concurrent.ExecutionException: java.util.ConcurrentModificationException
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at javax.swing.SwingWorker.get(SwingWorker.java:602)
	at com.dmdirc.addons.ui_swing.components.LoggingSwingWorker.done(LoggingSwingWorker.java:58)
	at javax.swing.SwingWorker$5.run(SwingWorker.java:737)
	at javax.swing.SwingWorker$DoSubmitAccumulativeRunnable.run(SwingWorker.java:832)
	at sun.swing.AccumulativeRunnable.run(AccumulativeRunnable.java:112)
	at javax.swing.SwingWorker$DoSubmitAccumulativeRunnable.actionPerformed(SwingWorker.java:842)
	at javax.swing.Timer.fireActionPerformed(Timer.java:313)
	at javax.swing.Timer$DoPostEvent.run(Timer.java:245)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:744)
	at java.awt.EventQueue.access$400(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:697)
	at java.awt.EventQueue$3.run(EventQueue.java:691)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:714)
	at com.dmdirc.addons.ui_swing.DMDircEventQueue.dispatchEvent(DMDircEventQueue.java:68)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
Caused by: java.util.ConcurrentModificationException
	at java.util.ArrayList$Itr.checkForComodification(ArrayList.java:886)
	at java.util.ArrayList$Itr.next(ArrayList.java:836)
	at com.google.common.collect.AbstractMapBasedMultimap$Itr.next(AbstractMapBasedMultimap.java:1150)
	at com.google.common.collect.TransformedIterator.next(TransformedIterator.java:48)
	at java.lang.Iterable.forEach(Iterable.java:74)
	at com.dmdirc.config.ConfigManager.removeListener(ConfigManager.java:438)
	at com.dmdirc.FrameContainer.setIcon(FrameContainer.java:293)
	at com.dmdirc.Server.updateIcon(Server.java:599)
	at com.dmdirc.Server.<init>(Server.java:227)
	at com.dmdirc.ServerFactoryImpl.getServer(ServerFactoryImpl.java:93)
	at com.dmdirc.ServerManager.createServer(ServerManager.java:111)
	at com.dmdirc.ServerManager.createServer(ServerManager.java:61)
	at com.dmdirc.commandparser.commands.global.NewServer.execute(NewServer.java:94)
	at com.dmdirc.commandparser.parsers.GlobalCommandParser.executeCommand(GlobalCommandParser.java:96)
	at com.dmdirc.commandparser.parsers.CommandParser.parseCommand(CommandParser.java:156)
	at com.dmdirc.commandparser.parsers.CommandParser.parseCommand(CommandParser.java:280)
	at com.dmdirc.commandparser.aliases.AliasCommandHandler.execute(AliasCommandHandler.java:50)
	at com.dmdirc.commandparser.parsers.GlobalCommandParser.executeCommand(GlobalCommandParser.java:96)
	at com.dmdirc.commandparser.parsers.CommandParser.parseCommand(CommandParser.java:156)
	at com.dmdirc.commandparser.parsers.CommandParser.parseCommand(CommandParser.java:280)
	at com.dmdirc.ui.input.InputHandler.enterPressed(InputHandler.java:589)
	at com.dmdirc.addons.ui_swing.components.inputfields.SwingInputHandler$5$1.doInBackground(SwingInputHandler.java:239)
	at javax.swing.SwingWorker$1.call(SwingWorker.java:295)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at javax.swing.SwingWorker.run(SwingWorker.java:334)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:744)
```

--- a/src/com/dmdirc/config/ConfigManager.java
+++ b/src/com/dmdirc/config/ConfigManager.java
@@ -435,7 +435,8 @@ class ConfigManager implements ConfigChangeListener, ConfigProviderListener,
     @Override
     public void removeListener(final ConfigChangeListener listener) {
         synchronized (listeners) {
-            listeners.keys().forEach(k -> listeners.remove(k, listener));
+            final Iterable<String> keys = new HashSet<>(listeners.keySet());
+            keys.forEach(k -> listeners.remove(k, listener));
         }
     }
 

